platform/ios/src/MGLMapView.mm

```diff
@@ -4759,7 +4804,11 @@ - (void)cameraIsChanging {
     }
 
     [self updateCompass];
-
+    
+    if (!self.scaleBarView.hidden) {
+        [(MGLScaleBarView *)self.scaleBarView setMetersPerPoint:[self metersPerPointAtLatitude:self.centerCoordinate.latitude]];
```

Comment:
Sorry, something isn’t clicking for me: if the developer hides the scale bar using `mapView.scaleView.hidden = YES`, then the user zooms way in, then the developer sets `mapView.scaleView.hidden = NO`, wouldn’t the scale bar appear with the wrong scale until the user starts zooming again?